lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +66,173 @@ def __init__(self, config):
+        return putative_int
 
-        # 2) construct the input directory staging area
-        input_directory = self.stage_input(params)
+    def _import_genbank_mass(self, params):
 
-        # 3) update default params
-        self.default_params.update(params)
-        params = self.default_params
-        self.generate_parents = params.get('generate_missing_genes')
-        self.generate_ids = params.get('generate_ids_if_needed')
-        if params.get('genetic_code'):
-            self.code_table = params['genetic_code']
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+        genome_objs = []
 
-        # 4) Do the upload
-        files = self._find_input_files(input_directory)
-        consolidated_file = self._join_files_skip_empty_lines(files)
-        genome = self.parse_genbank(consolidated_file, params)
-        if params.get('genetic_code'):
-            genome["genetic_code"] = params['genetic_code']
+        for idx, input_params in enumerate(inputs):
 
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
+            genome_obj = _Genome()
+            genome_obj.genome_name = input_params['genome_name']
 
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
+            # construct the input directory staging area
+            genome_obj.input_directory = self._stage_input(input_params)
 
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
+            # update default params
+            input_params = {**self.default_params, **input_params}
+            genome_obj.genome_meta = input_params['metadata']
+            inputs[idx] = input_params
+
+            genome_obj.genome_type = input_params.get('genome_type', 'isolate')
+            genome_obj.generate_parents = input_params.get('generate_missing_genes')
+            genome_obj.generate_ids = input_params.get('generate_ids_if_needed')
+            genome_obj.use_existing_assembly = input_params.get('use_existing_assembly')
+            if input_params.get('genetic_code'):
+                genome_obj.code_table = input_params['genetic_code']
+
+            # find genbank file
+            files = self._find_input_files(genome_obj.input_directory)
+            genome_obj.consolidated_file = self._join_files_skip_empty_lines(files)
+
+            # gather all objects
+            genome_objs.append(genome_obj)
+
+        self._get_contigs_and_validate_existing_assembly(genome_objs)
+
+        self._save_files_to_blobstore_and_set_handle_service_output(genome_objs)
+
+        self._save_assemblies(workspace_id, genome_objs)
```

Comment:
Similar to the parse_genomes thing, save files to blobstore is saving the genbank files before the assembly data has been validated in _save_assemblies

Reply:
ah, I thought we have to save files to blobstore before saving any assemblies. Is this not true anymore?

Reply:
That's true. You don't have to save the files to the blobstore before you validate the assembly data though

Reply:
I see what you are saying now. 
If parse_genomes step is moved before any upload, this would not be a problem anymore.

Reply:
All I want is to not upload data and then find there's a problem and we can't finish. Whether than means using current methods as is, moving chunks of them around, or making new methods is up to you

Reply:
This will be fixed in the move parse_genbank PR

Reply:
:+1: